queue/queue.go

```diff
@@ -13,81 +15,56 @@ import (
 	"github.com/streadway/amqp"
 )
 
-func (qm *Manager) setupLogging() error {
-	logFileName := fmt.Sprintf("/var/log/temporal/%s_serice.log", qm.QueueName)
+func (qm *Manager) setupLogging(logFilePath ...string) error {
+	var logFileName string
+	if len(logFilePath) > 0 {
+		logFileName = fmt.Sprintf("%s-%s_service.log", logFilePath[0], qm.QueueName)
+	} else {
+		logFileName = fmt.Sprintf("/var/log/temporal/%s_service.log", qm.QueueName)
+	}
 	logFile, err := os.OpenFile(logFileName, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0640)
 	if err != nil {
 		return err
 	}
 	logger := log.New()
 	logger.Out = logFile
-	qm.Logger = logger
-	qm.Logger.Info("Logging initialized")
-	return nil
-}
-
-func (qm *Manager) parseQueueName(queueName string) error {
-	host, err := os.Hostname()
-	if err != nil {
-		return err
-	}
-	qm.QueueName = fmt.Sprintf("%s+%s", host, queueName)
+	qm.logger = logger
+	qm.logger.Info("Logging initialized")
 	return nil
 }
 
 // Initialize is used to connect to the given queue, for publishing or consuming purposes
-func Initialize(queueName, connectionURL string, publish, service bool) (*Manager, error) {
+func Initialize(queueName, connectionURL string, publish, service bool, logFilePath ...string) (*Manager, error) {
 	conn, err := setupConnection(connectionURL)
 	if err != nil {
 		return nil, err
 	}
-	qm := Manager{Connection: conn}
+	qm := Manager{connection: conn}
+	// open a channel
 	if err := qm.OpenChannel(); err != nil {
```

Comment:
`OpenChannel` is only ever called here, this should be private

Reply:
I think the function should be done inline actually - it might be best to instantiate as much of `qm` as possible in one go, instead of doing it piecewise - one thing I like to do is gather all dependencies, and as late as possible, create the relevant class:

```go
return &Manager{ /* a bunch of stuff */ }, nil
```

Reply:
I was able to re-organize the order of calls for that function, however `Manager` can only be instantiated with the string variables, the rest of the setup requires invocation in a particular order.